### PR TITLE
Reformat and flake8 e203 rule

### PR DESCRIPTION
**Pull request by croumegous**
MR with little reformatting and minor code quality improvement (minus 13 lines of code).
As well as applying flake E203 rule: `var : type = value` becomes `var: type = value`.

Separated into two different commits for easier review

### CODE REVIEW

File: tinygrad/codegen/ast.py [hunk 1]
```diff
@@ -55,7 +55,7 @@ def __init__(self, ast:LazyOp, output_buffer=None):
           break
 
     # create the buffer we are returning (as the same type as the input buffers) and add it as the first buffer
-    self.ret = output_buffer if output_buffer else type(self.bufs[0])(output_shape if output_shape else self.info.shape, force_create=True)
+    self.ret = output_buffer or type(self.bufs[0])(output_shape or self.info.shape, force_create=True)
```
Comment (geohot) [inline tinygrad/codegen/ast.py:58]: Please do one thing in this commit. If this introduces a bug, I'll never find it.

Comment (croumegous) [reply tinygrad/codegen/ast.py]: rebased with changes of https://github.com/geohot/tinygrad/pull/673 and reverted all formatting that could create buggy corner case.

File: tinygrad/codegen/ast.py [hunk 2]
```diff
@@ -156,10 +156,8 @@ def simplify_merge_adjacent(self):
     # TODO: move this into shapetracker, with tests!
     rets = [[(shapes[j][0], strides[j][0])] for j in range(len(shapes))]
     for i in range(1, len(shapes[0])):
-      can_merge = []
-      for j in range(len(shapes)):
-        # TODO: added the always mergeability of 1s, is this right? if so, add to shapetracker in the 1 case
-        can_merge.append((strides[j][i] != 0 and rets[j][-1][1] == shapes[j][i]*strides[j][i]) or (strides[j][i] == 0 and rets[j][-1][1] == 0))
+      # TODO: added the always mergeability of 1s, is this right? if so, add to shapetracker in the 1 case
+      can_merge = [((strides[j][i] != 0 and rets[j][-1][1] == shapes[j][i] * strides[j][i]) or (strides[j][i] == 0 and rets[j][-1][1] == 0)) for j in range(len(shapes))]
```
Comment (geohot) [inline tinygrad/codegen/ast.py:160]: NO!

**Comment by geohot** [comment]
Like I said, you can't combine formatting changes with other things!!

I know you think they don't change functionality, but would you bet $10,000 on that? I make mistakes where I think things are equivalent all the time.

File: tinygrad/codegen/gpu.py [hunk 3]
```diff
@@ -142,13 +142,12 @@ def load(self, buf_index:int, idx_override:Optional[str]=None) -> List[Token]:
   def ast_parse(self, x, acc:List[Token], do_reduce=False) -> List[Token]:
     if not isinstance(x, LazyOp): return self.load(self.bufs.index(x), "mid" if x is None else None)  # hack for local
     if isinstance(x.op, ReduceOps) and not do_reduce: return acc
-    values : List[List[Token]] = ([acc] if isinstance(x.op, ReduceOps) else []) + [self.ast_parse(v, acc, do_reduce) for v in x.src]
+    values: List[List[Token]] = ([acc] if isinstance(x.op, ReduceOps) else []) + [self.ast_parse(v, acc, do_reduce) for v in x.src]
     code = GPUCodegen.code_for_op[x.op]  # TODO: replace this with a function
-    if len(values) == 2:
```
Comment (geohot) [inline tinygrad/codegen/gpu.py:147]: ?

Comment (croumegous) [reply tinygrad/codegen/gpu.py:147]: This doesn't change the behaviour, I swear.

here is the change   
before :
```python
if len(values) == 2:
	return "behaviour len(values) == 2"
else:
	return "behaviour lend(values) != 2"
```	
	
Now :
```python
if len(values) != 2: return "behaviour lend(values) != 2"
return "behaviour len(values) == 2"
```
If you are afraid those changes can affect the proper functioning of tinygrad for no real added value I can ditch the format commit and only keep the flake E203 one.

Sorry for the inconvenience.

**Comment by geohot** [comment]
I thought I made that clear. If you want to resubmit, submit with *only* the E203 change and not a single other line different.
